### PR TITLE
4 can slot

### PR DESCRIPTION
**Pull request by Sinjhin**
Added functionality for `<can-slot>` elements to be replaced by `<can-template>` elements using default scope.

Depends on https://github.com/canjs/can-stache/pull/209

### CODE REVIEW

File: can-component.js [hunk 1]
```diff
@@ -246,29 +251,17 @@ var Component = Construct.extend(
...
-				}
-
-				// We need be alerted to when a <content> element is rendered so we can put the original contents of the widget in its place
-				options.tags.content = function contentHookup(el, contentTagData) {
-					// First check if there was content within the custom tag
-					// otherwise, render what was within <content>, the default code.
-					// `componentTagData.subtemplate` is the content inside this component
-					var subtemplate = componentTagData.subtemplate || contentTagData.subtemplate,
+			// Returns a hookupFuction that gets the proper template, renders it, and adds it to nodeLists
+			makeHookup = function(tagName, getPrimaryTemplate) {
```
Comment (justinbmeyer) [inline can-component.js]: can this function be pulled out into the modules scope?

File: can-component.js [hunk 2]
```diff
@@ -307,18 +300,39 @@ var Component = Construct.extend(
...
+			}
+
+			// If this component has a view (that we've already converted to a renderer)
+			if (this.constructor.renderer) {
+				// If `options.tags` doesn't exist set it to an empty object.
+				if (!options.tags) {
+					options.tags = {};
+				}
+
+				if (!isEmptyObject(templates)) {
```
Comment (justinbmeyer) [inline can-component.js]: hmmm, eventually, we probably should support dynamic templates.   No need for now.

File: docs/can-slot.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+@typedef {can-stache.sectionRenderer} can-component/can-slot <can-slot>
```
Comment (justinbmeyer) [inline docs/can-slot.md]: does this show it can have default content?

File: docs/can-slot.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+@parent can-component.elements
+
+@description Positions `<can-template />` elements with matching name attribute using current scope.
+
+@signature `<can-slot name='templateName' />`
```
Comment (justinbmeyer) [inline docs/can-slot.md]: The params aren't detailed.  We'd normally CAPITALIZE `templateName` and specify that as a param.

File: docs/can-slot.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	}
+});
+
+Component.extend({
+	tag : 'my-email',
+	view : stache(
+		'<can-slot name="subject" />'
+	),
+	ViewModel,
+	leakScope: true
```
Comment (justinbmeyer) [inline docs/can-slot.md]: We probably shouldn't show a `leakScope` example.

File: test/component-slot-test.js [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+		'</my-email>'
+	);
+
+	var testView = renderer();
+
+	equal(testView.firstChild.childNodes[0].textContent, 'Hello World');
+	equal(testView.firstChild.childNodes[1].textContent, 'Hello World');
+});
+
+// test("<can-slot> works with <content>", function() {
```
Comment (justinbmeyer) [inline test/component-slot-test.js]: commented out?

File: test/component-slot-test.js [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+	});
+
+	Component.extend({
+		tag : 'my-email',
+		view : stache(
+			'<can-slot name="subject" />' +
+			'<can-slot name="subject" />'
+		),
+		ViewModel,
+		leakScope: true
```
Comment (justinbmeyer) [inline test/component-slot-test.js]: we need tests w/o leakScope

**Comment by Sinjhin** [comment]
Moved to branch `4-can-slot-2` for this feature.

**Pull request closed without merging**